### PR TITLE
Adding support for loading authConfig from dockers config

### PR DESCRIPTION
**Pull request by brujoand**
My Java fu is a bit outdated, but this should get the job done.
Thanks

### CODE REVIEW

**Comment by brujoand** [comment]
Any feedback on this?

**Comment by davidxia** [comment]
@brujoand Sorry for the delayed response. I just merged https://github.com/spotify/docker-client/pull/261 but haven't gotten a chance to see go over how yours differs from that one yet.

**Comment by mattnworb** [comment]
Overall LGTM, but what happens if there are multiple servers in the `auths` element of `~/.docker/config.json`? Seems like only the first one is used. It might be best to attempt to match it up with a user-supplied URL.

File: src/main/java/com/spotify/docker/client/messages/AuthConfig.java [hunk 1]
```diff
@@ -168,6 +178,69 @@ public String serverAddress() {
...
+    }
+
+    private JsonNode extractDockerJson(String configFile) throws IOException {
+      ObjectMapper mapper = new ObjectMapper();
+      JsonNode config = mapper.readTree(new File(configFile));
+
+      if (config.has("auths")) {
+        return config.get("auths");
+      }
+      return config;
```
Comment (mattnworb) [inline src/main/java/com/spotify/docker/client/messages/AuthConfig.java]: I don't think there would be any value in returning the JsonNode if it doesn't have an `auths` field, as the calling code takes the first field in the return value as the servername

**Comment by mattnworb** [comment]
Oh and thank you for adding tests with your change :wink:

**Comment by brujoand** [comment]
Thanks for the feedback, I've cleaned up some of this on the branch we use at work. I'll push the changes when I get in tomorrow! :)

**Comment by brujoand** [comment]
I moved this to #266
So I'll just close this one

**Pull request closed without merging**